Watchman.Discord/Areas/Administration/Controllers/AdministrationController.cs

```diff
@@ -87,34 +87,74 @@ public async Task ReadUserMessages(DiscordRequest request, Contexts contexts)
+
+            if (args.Count() < 2)
             {
                 throw new NotEnoughArgumentsException();
             }
 
-            var roleName = args[0].Value;
-            var toSetAsSafe = args[1].Value == "safe";
-
-            var serverRole = _usersRolesService.GetRoleByName(roleName, contexts.Server);
-            if (serverRole == null)
+            if (args.HasDuplicates())
             {
-                throw new RoleNotFoundException(roleName);
+                throw new ArgumentsDuplicatedException();
             }
 
-            if (toSetAsSafe)
+            var isSafeArgument = args.Last().Value.ToLower();
+
+            if (isSafeArgument != "safe"
+                && isSafeArgument != "unsafe")
             {
-                var command = new SetRoleAsSafeCommand(roleName, contexts.Server.Id);
-                await _commandBus.ExecuteAsync(command);
+                throw new NotEnoughArgumentsException();
             }
-            else
+
+            var isSafe = isSafeArgument == "safe" ? true : false;
+
+            var safeRoles = this._queryBus.Execute(new GetDiscordServerSafeRolesQuery(contexts.Server.Id)).SafeRoles;
+            var argRoleNames = args.Select(x => x.Value).SkipLast(1).ToList();
+
+            foreach (var roleNameToCheck in argRoleNames)
             {
-                var command = new SetRoleAsUnsafeCommand(roleName, contexts.Server.Id);
-                await _commandBus.ExecuteAsync(command);
+                var serverRole = _usersRolesService.GetRoleByName(roleNameToCheck, contexts.Server);
+
+                if (serverRole == null)
+                {
+                    throw new RoleNotFoundException(roleNameToCheck);
+                }
+
+                if (isSafe
+                    && safeRoles.Select(obj => obj.Name)
+                    .Where(x => x == roleNameToCheck).Select(x => x).Count() != 0)
+                {
+                    throw new RoleIsSafeAlreadyException(roleNameToCheck);
+                }
+                else if (!isSafe
+                    && safeRoles.Select(obj => obj.Name)
+                    .Where(x => x == roleNameToCheck).Select(x => x).Count() == 0)
+                {
+                    throw new RoleIsUnsafeAlreadyException(roleNameToCheck);
+                }
             }
 
-            var messagesService = _messagesServiceFactory.Create(contexts);
-            await messagesService.SendResponse(x => x.RoleSettingsChanged(roleName), contexts);
+            foreach (var roleNameToSet in argRoleNames)
+            {
+                if (isSafe)
+                {
+                     await _commandBus.ExecuteAsync(
+                        new SetRoleAsSafeCommand(roleNameToSet, contexts.Server.Id)
+                        );
+                }
+                else
+                {
+                    await _commandBus.ExecuteAsync(
```

Comment:
czemu to jest w nowej linii?
jeśli brakuje ci miejsca na monitorze, to wrzuć komende do osobnej zmiennej